src/proposals/20251002/SparkEthereum_20251002.t.sol

```diff
@@ -0,0 +1,504 @@
+
+        _testSlashingIsDisabledUnlessMiddlewareIsSet();
+    }
+
+    function _testSlashingIsDisabledUnlessMiddlewareIsSet() public {
+        address alice      = makeAddr("alice");
+        address bob        = makeAddr("bob");
+        address NETWORK    = Ethereum.SPARK_PROXY;
+        address OPERATOR   = Ethereum.SPARK_PROXY;
+        address MIDDLEWARE = makeAddr("middleware");
+ 
+        IERC20 spk           = IERC20(Ethereum.SPK);
+        IStakedSPK stSpk     = IStakedSPK(Ethereum.STSPK);
+        IVetoSlasher slasher = IVetoSlasher(VETO_SLASHER);
+
+        INetworkMiddlewareService middlewareService = INetworkMiddlewareService(slasher.NETWORK_MIDDLEWARE_SERVICE());
+
+        bytes32 subnetwork = bytes32(uint256(uint160(NETWORK)) << 96 | 0);  // Subnetwork.subnetwork(network, 0)
+
+        uint256 ACTIVE_STAKE = stSpk.activeStake();
+
+        // --- Step 1: Deposit 10m SPK to stSPK as two users
+
+        deal(address(spk), alice, 6_000_000e18);
+        deal(address(spk), bob,   4_000_000e18);
+
+        vm.startPrank(alice);
+        spk.approve(address(stSpk), 6_000_000e18);
+        stSpk.deposit(alice, 6_000_000e18);
+        vm.stopPrank();
+
+        vm.startPrank(bob);
+        spk.approve(address(stSpk), 4_000_000e18);
+        stSpk.deposit(bob, 4_000_000e18);
+        vm.stopPrank();
+
+        uint48 depositTimestamp = uint48(block.timestamp);
+
+        skip(24 hours);  // Warp 24 hours
+
+        // --- Step 2: Request a slash of all staked SPK (show that network limit is hit)
+
+        uint48 captureTimestamp = uint48(block.timestamp - 1 seconds);  // Can't capture current timestamp and above
+
+        // Demonstrate that the slashable stake increases with new deposits
+        assertEq(slasher.slashableStake(subnetwork, OPERATOR, depositTimestamp - 1, ""), 0);
+        assertEq(slasher.slashableStake(subnetwork, OPERATOR, depositTimestamp,     ""), ACTIVE_STAKE + 10_000_000e18);
+        assertEq(slasher.slashableStake(subnetwork, OPERATOR, captureTimestamp,     ""), ACTIVE_STAKE + 10_000_000e18);
+
+        // There is no middleware, so slashing is impossible
+        assertEq(middlewareService.middleware(NETWORK), address(0));
+
+        vm.prank(NETWORK);
+        vm.expectRevert(NotNetworkMiddleware.selector);
+        slasher.requestSlash(subnetwork, OPERATOR, 10_000_000e18, captureTimestamp, "");
+
+        // Show how it would work if middleware was set
+        vm.prank(NETWORK);
+        middlewareService.setMiddleware(MIDDLEWARE);
+
+        // Its now possible
+        assertEq(middlewareService.middleware(NETWORK), MIDDLEWARE);
+
+        vm.prank(MIDDLEWARE);
+        uint256 slashIndex = slasher.requestSlash(subnetwork, OPERATOR, 10_000_000e18, captureTimestamp, "");
+
+        skip(3 days + 1);
+
+        vm.prank(MIDDLEWARE);
+        slasher.executeSlash(slashIndex, "");
+    }
+
+    function test_ETHEREUM_sparkLend_withdrawUsdsDaiReserves() public onChain(ChainIdUtils.Ethereum()) {
```

Comment:
Can we please move this between E3 ("SparkLend - Reduce Stablecoin Market Reserve Factors") and E5 ("Spark Savings v2 - Launch Savings Vaults for USDC, USDT, and ETH") as it is in the post?